src/pQueue.js

```diff
@@ -0,0 +1,37 @@
+'use strict'
+export default class pQueue {
+  constructor() {
+    this.q = []
+  }
+
+  enqueue(element, priority){
+    // push an object to the end of the arrat
+    // sort the array according to priority
+    this.q[this.q.length] = {element, priority}
+
+    var sorted = this.q.sort(function(a,b){
+      return parseFloat(b.priority) - parseFloat(a.priority)
+    })
+    return sorted
+  }
+
+  front() {
+    return this.q.length > 0 ? this.q[0] : null
+  }
+
+  back() {
+    return this.q.length > 0 ? this.q[this.q.length - 1] : null
+  }
+
+  dequeue() {
+    return this.q.length > 0 ? this.q.shift() : null
+  }
+
+  isEmpty() {
+    return this.q.length <= 0
+  }
+
+  lengths() {
+    return this.q.length 
+  }
+}
```

Comment:
Everything looks good to me. But like I said to y'all in person. I think the built .length method is cheating. You should try and implement your own. But if you understand then go ahead and merge.